Lib/Stack.hpp

```diff
@@ -304,22 +373,7 @@ class Stack
    * @since 11/03/2006 Bellevue
    */
   inline
-  void push(const C& elem)
-  {
```

Comment:
Just curious: did you realize neither reference nor rvalue reference make sense in this context?

Reply:
Yes exactly. So if we take  a regular value `C` instead of a reference, then it's up to the caller whether they wanna move or copy into the function, while if we use `C const&` we force the caller to copy and if we use `C&&` we force the caller to move.